sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/models/ChangeFeedProcessorItem.java

```diff
@@ -55,8 +60,18 @@ public JsonNode getPrevious() {
      * @return change feed metadata.
      */
     @Beta(value = Beta.SinceVersion.V4_37_0, warningText = Beta.PREVIEW_SUBJECT_TO_CHANGE_WARNING)
+    @JsonIgnore
     public ChangeFeedMetaData getChangeFeedMetaData() {
-        return changeFeedMetaData;
+
+        if (this.changeFeedProcessorItemAsJsonNode == null) {
+            this.changeFeedProcessorItemAsJsonNode = constructChangeFeedProcessorItemAsJsonNode();
+        }
+
+        if (this.changeFeedMetaDataInternal == null) {
+            this.changeFeedMetaDataInternal = Utils.getSimpleObjectMapper().convertValue(this.changeFeedMetaData, ChangeFeedMetaData.class);
```

Comment:
do we need to validate changeFeedMetaData is not null?

Reply:
`convertValue()` will return `null` with no exceptions. So, no additional check needed.